### PR TITLE
Update page with template examples

### PR DESCRIPTION
**Pull request by TheFes**
Hi,

I've taken the liberty to make updates to your template page.
Note that I'm not finished yet, I will continue when I find some more time.

I've rewritten the first part to the modern format, and also added some explanation on that.
I also corrected some templates which were not correct (as I also mentioned on Facebook a few days ago).

Also added a `unique_id` everywhere, and added an `availability` template where needed.

### CODE REVIEW

**Comment by TheFes** [comment]
Marked as draft for now, as I'm still working on it

File: homeassistant/homeassistant_templates.md [hunk 1]
```diff
@@ -313,19 +349,20 @@ Count the days before the paper bin will be picked up.
...
+        state: >
+          {% set pickup_date = state_attr('sensor.cyclus_papier','sort_date') | as_datetime %}
+          {{ (pickup_date.date() - now().date()).days }}
+        unit_of_measurement: days
+        availability: "{{ state_attr('sensor.cyclus_papier','sort_date') | as_datetime(default=none) is not none }}"
 {% endraw %}
 ```
 
-The sensor `sensor.cyclus_papier` has an attribute `Sort_date` which holds the date for the bin pickup day in the format "YYYYMMDD".\
+The sensor `sensor.cyclus_papier` has an attribute `sort_date` which holds the date for the bin pickup day in the format "YYYY-MM-DD".\
```
Comment (vdbrink) [inline homeassistant/homeassistant_templates.md:365]: The original text was correct 
image

**Comment by vdbrink** [comment]
@TheFes Thanks for your input!
Let me know when you're finished then I can review the PR.

File: homeassistant/homeassistant_templates.md (hunk 1 above)
Comment (TheFes) [reply homeassistant/homeassistant_templates.md]: I've changed it to use `strptime`

**Comment by TheFes** [comment]
Done

**Comment by TheFes** [comment]
Sorry, made a mistake with the T-shirt one
image

forgot to put everything under `- sensor:` there. Maybe you can add it yourself, otherwise I can create a PR for it.

**Comment by TheFes** [comment]
It should have been:
```yaml
# Sourcecode by vdbrink.github.io
# configuration.yaml
template:
  - sensor:
      - name: "Wear outside"
        unique_id: sensor_wear_outside
        icon: mdi:tshirt-crew
        state: >-
          {% set temp = states('sensor.tempest_temperature_feels_like_rounded') | float %}
          {% if temp <= 5 %}
            winter jacket and hand gloves
          {% elif temp <= 14 %}
            softshell
          {% elif temp <= 18 %}
            thin jacket
          {% else %}
            T-shirt
          {% endif %}
    availability: "{{ 'sensor.tempest_temperature_feels_like_rounded' | has_value }}
```

**Comment by vdbrink** [comment]
I updated the page with your fix

**Pull request merged**